docs/configure-artifact-repository.md

```diff
@@ -107,43 +110,176 @@ $ cat > policy.json <<EOF
+
+```bash
+aws iam create-user --user-name $mybucket-user
+aws iam put-user-policy --user-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+aws iam create-access-key --user-name $mybucket-user > access-key.json
+```
+
+You can then use the access keys by configuring your artifact as below:
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      key: path/in/bucket/my-output-artifact.tgz
+      # The following secret entries are secret selectors.
+      # They reference the k8s secret named 'my-s3-credentials'.
+      # This secret is expected to have the keys 'accessKey' and 'secretKey',
+      # containing the base64 encoded credentials to the bucket.
+      accessKeySecret:
+        name: my-s3-credentials
+        key: accessKey
+      secretKeySecret:
+        name: my-s3-credentials
+        key: secretKey
+```
+
+### AWS S3 with IAM role credentials
+
+You can use IAM role credentials for temporary access.
+
+If you have created a bucket and policy as described above, you can then create an IAM role and add the policy with:
+
+```bash
+aws iam create-role --role-name $mybucket-role
+aws iam put-role-policy --role-name $mybucket-user --policy-name $mybucket-policy --policy-document file://policy.json
+```
+
+You can then obtain the credentials set by running:
+
+```bash
+aws sts assume-role --role-arn rn:aws:iam::012345678901:role/$mybucket-role
+```
+
+After which you can use the credentials in your artifact configuration:
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      key: path/in/bucket/my-output-artifact.tgz
+      # The following secret entries are secret selectors.
+      # They reference the k8s secret named 'my-s3-credentials'.
+      # This secret is expected to have the keys 'accessKey', 'secretKey', and 'sessionToken',
+      # containing the base64 encoded credentials to the bucket.
+      accessKeySecret:
+        name: my-s3-credentials
+        key: accessKey
+      secretKeySecret:
+        name: my-s3-credentials
+        key: secretKey
+      sessionTokenSecret:
+        name: my-s3-credentials
+        key: sessionToken
+```
+
+!!! Note "IAM role credentials are temporary"
+    IAM role credentials are temporary, so you must refresh them periodically via an external mechanism.
+    Argo Workflows will not refresh them for you.
+
+### AWS S3 Access Grants
+
+You can use [S3 Access Grants](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-grants.html) for temporary, reduced scope access.
+Read the [getting started guide](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-grants-get-started.html) to set this up in your AWS account.
```

Comment:
Ah I see, you put this here because there's no commands to run in this section, unlike the others (as it's more complicated. Let's re-word this a bit then:

```suggestion
Follow the [AWS guide](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-grants-get-started.html) to set this up in your account and retrieve access grant credentials.
```

- This now segues into the next sentence more neatly
- also "getting started guide" is ambiguous since Argo has a similarly named guide